daemon/src/surface.rs

```diff
@@ -188,26 +207,31 @@ impl Surface {
                 }
             }
             loop {
-                let files: Vec<PathBuf> = WalkDir::new(path)
-                    .into_iter()
-                    .filter_map(|e| e.ok())
-                    .filter(|e| {
-                        if let Some(guess) = new_mime_guess::from_path(e.path()).first() {
-                            guess.type_() == "image"
-                        } else {
-                            false
+                let is_zero = self.current_index == 0;
+                let is_below_len =
+                    !self.image_paths.is_empty() && self.current_index < self.image_paths.len();
+
+                let img_path = match (is_zero, is_below_len) {
+                    // Use
+                    (false, true) => self.image_paths[self.current_index].clone(),
+                    (true, true) => self.image_paths[0].clone(),
+                    _ => {
+                        let files = self.get_image_files_from_dir(path);
+                        // There are no images, forcefully break out of the loop
+                        if files.is_empty() {
+                            bail!("Directory {path:?} is empty");
                         }
-                    })
-                    .map(|e| e.path().to_path_buf())
-                    .collect();
-
-                if files.is_empty() {
-                    // There are no images, forcefully break out of the loop
-                    // modulus operator panics when the right argument is 0
-                    bail!("Directory {path:?} is empty");
-                }
 
-                let img_path = files[rand::random::<usize>() % files.len()].clone();
+                        let img_path = files[rand::random::<usize>() % files.len()].clone();
+
+                        self.image_paths.push(img_path.clone());
```

Comment:
In this case we are also saving images that cannot be opened. I have added this loop because it can happen that some images cannot be opened (for example in #21 it's a library issue). By saving this images here I feel like `previous-wallpaper` command wouldn't work as expected.

Reply:
I see, so instead I should save it within that `Ok(image)` block after checking if it already exists in `img_paths`?